helix-term/src/commands/typed.rs

```diff
@@ -1410,47 +1410,51 @@ fn tutor(
     Ok(())
 }
 
+fn abort_goto_line_number_preview(cx: &mut compositor::Context) {
+    if let Some(last_selection) = &cx.editor.last_selection {
+        let scrolloff = cx.editor.config().scrolloff;
+        let (view, doc) = current!(cx.editor);
+        doc.set_selection(view.id, last_selection.clone());
+        view.ensure_cursor_in_view(doc, scrolloff);
+        cx.editor.last_selection = None;
+    }
+}
+
 pub(super) fn goto_line_number(
     cx: &mut compositor::Context,
     args: &[Cow<str>],
     event: PromptEvent,
 ) -> anyhow::Result<()> {
     match event {
-        PromptEvent::Abort => {
-            if let Some(line_number) = cx.editor.last_line_number {
-                goto_line_impl(cx.editor, NonZeroUsize::new(line_number));
-                let (view, doc) = current!(cx.editor);
-                view.ensure_cursor_in_view(doc, line_number);
-                cx.editor.last_line_number = None;
-            }
-            return Ok(());
-        }
+        PromptEvent::Abort => abort_goto_line_number_preview(cx),
         PromptEvent::Validate => {
             ensure!(!args.is_empty(), "Line number required");
-            cx.editor.last_line_number = None;
+
+            if let Some(last_selection) = &cx.editor.last_selection {
+                let (view, doc) = current!(cx.editor);
+                view.jumps.push((doc.id(), last_selection.clone()));
```

Comment:
good idea about reusing `last_selection` here to get around the fact that we might currently be at a different preview :+1: